### ISSUE TITLE
Check Permissions in a generic way 

### ISSUE THREAD

**Issue by matthias-hiemer** [eadwinCode/django-ninja-extra#52]
Hi there!

I'm new to Ninja Extra and was wondering how I can check for permissions in a generic way.
Checking a _single_ permission class of type BasePermission works fine to protect an endpoint.

For Example:
```
    class HasViewDocumentPermission(permissions.BasePermission):

        def has_permission(self, request, controller) -> bool:
            user = request.user or request.auth
            return user and user.has_perm("api.view_document")

    @http_get(
        "",
        response={200: List[DocumentSchema], 404: Message},
        permissions=[HasViewDocumentPermission],
        by_alias=True,
    )
    @requires()
    def get_documents(self):
        # Gets documents ...
```

But: We have a lot of permissions in multiple groups in our project. So, creating dozens of boilerplate permission classes doesn't seem to me that I am on the right track.  
Can I do this somehow by using a BasePermission class, that takes a permission as parameter? Or do I need another concept, like another decorator? 

Thanks and best regards
Matthias

**Comment by eadwinCode** [comment]
@matthias-hiemer, yes you can modify the BasePermission class to accept permissions but for me to really suggest a different concept to follow, I need a good example of your permission system. 
Also if you are using group permissions, you can actually assert at group level too instead of asserting individual permissions in a group

**Comment by matthias-hiemer** [comment]
@eadwinCode 
Using hard-coded groups  in the controllers would mean we couldn't make use of model permissions. For example, we want to define or re-configure permissions of groups while the system is running - without re-building the backend.

Basically we want to check the model permissions that Django generates for every endpoint.
The endpoint GET /api/document should only be available for users with the permission "view_document".
The endpoint DELETE /api/document should only be available for users with the permission "delete_document".
POST /api/document only with "add_document", and so on.


I would love to have something like this: 
`    
@http_get(
        "",
        response={200: List[DocumentSchema], 404: Message},
       permissions=[ModelPermission('view_document')],
        by_alias=True,
)
def get_documents(self):
      # Gets documents ...
`

**Comment by eadwinCode** [comment]
Alright, I will think about this and get something for you.

**Comment by eadwinCode** [comment]
@matthias-hiemer [v0.18.7](https://github.com/eadwinCode/django-ninja-extra/releases/tag/0.18.7) will solve this problem. Yeah?

**Comment by matthias-hiemer** [comment]
@eadwinCode 
That's amazing! Thanks you very much! It looks exactly like the solution we need. I will try to integrate it in our application as soon as possible 😃

**Comment by eadwinCode** [comment]
Alright cool.
